ui/app/services/breadcrumbs.js

```diff
@@ -32,14 +38,24 @@ export default class BreadcrumbsService extends Service {
       // Breadcrumbs are either an array of static crumbs
       // or a function that returns breadcrumbs given the current
       // model for the route's controller.
-      let breadcrumbs = route.breadcrumbs || [];
-      if (typeof breadcrumbs === 'function') {
-        breadcrumbs = breadcrumbs(route.get('controller.model')) || [];
-      }
+      let crumb = this.generateCrumb(route);
+      crumbs.push(...crumb);
+
+      return crumbs;
+    }, []);
```

Comment:
Mostly nitpicking/personal preference, but these new functions seem to be pure functions, so I maybe they would be better placed outside the `Service` object, just as helper functions in the same file. Personally I find a bit hard to read longer arrow functions set as attributes.

This could also help to write unit tests for them.